DNN Platform/DotNetNuke.ModulePipeline/ModuleControlPipeline.cs

```diff
@@ -0,0 +1,194 @@
+            return control;
+        }
+
+        public Control LoadModuleControl(TemplateControl containerControl, ModuleInfo moduleConfiguration)
+        {
+            if (TracelLogger.IsDebugEnabled)
+                TracelLogger.Debug($"ModuleControlFactory.LoadModuleControl Start (TabId:{moduleConfiguration.TabID},ModuleId:{moduleConfiguration.ModuleID}): ModuleControlSource:{moduleConfiguration.ModuleControl.ControlSrc}");
+            Control control = null;
+            IModuleControlFactory controlFactory = GetModuleControlFactory(moduleConfiguration.ModuleControl.ControlSrc);
+
+            if (controlFactory != null)
+            {
+                control = controlFactory.CreateModuleControl(containerControl, moduleConfiguration);
+            }
+
+            // set the control ID to the resource file name ( ie. controlname.ascx = controlname )
+            // this is necessary for the Localization in PageBase
+            if (control != null)
+            {
+                control.ID = Path.GetFileNameWithoutExtension(moduleConfiguration.ModuleControl.ControlSrc);
+
+                var moduleControl = control as IModuleControl;
+
+                if (moduleControl != null)
+                {
+                    moduleControl.ModuleContext.Configuration = moduleConfiguration;
+                }
+            }
+
+            if (TracelLogger.IsDebugEnabled)
+                TracelLogger.Debug($"ModuleControlFactory.LoadModuleControl End (TabId:{moduleConfiguration.TabID},ModuleId:{moduleConfiguration.ModuleID}): ModuleControlSource:{moduleConfiguration.ModuleControl.ControlSrc}");
+            return control;
+        }
+
+        public Control LoadSettingsControl(TemplateControl containerControl, ModuleInfo moduleConfiguration, string controlSrc)
+        {
+            if (TracelLogger.IsDebugEnabled)
+                TracelLogger.Debug($"ModuleControlFactory.LoadSettingsControl Start (TabId:{moduleConfiguration.TabID},ModuleId:{moduleConfiguration.ModuleID}): ModuleControlSource:{moduleConfiguration.ModuleControl.ControlSrc}");
+
+            Control control = null;
+            IModuleControlFactory controlFactory = GetModuleControlFactory(controlSrc);
+
+            if (controlFactory != null)
+            {
+                control = controlFactory.CreateSettingsControl(containerControl, moduleConfiguration, controlSrc);
+            }
+
+            // set the control ID to the resource file name ( ie. controlname.ascx = controlname )
+            // this is necessary for the Localization in PageBase
+            if (control != null)
+            {
+                var fileNameWithoutExtension = Path.GetFileNameWithoutExtension(controlSrc);
+                if (fileNameWithoutExtension != null)
+                {
+                    control.ID = fileNameWithoutExtension.Replace('.', '-');
```

Comment:
I'm noticing that we don't replace `.` with `-` in `LoadModuleControl` (line 110), anyone know it that's on purpose (or why we're replacing here)? @dnnsoftware/tag @dnnsoftware/tag-group @dnnsoftware/approvers

Reply:
This code is the exact same code copied from the old `ModuleControlFactory`. I am hesitant to make any changes that modifies the original code that was unchanged in this pull request. *I am in support of working through follow up issues. *

If I am in the minority here, we can make changes as needed.

Reply:
No, I completely agree that any issues raised with the copied code should be follow-up issues.

Reply:
Could you create an issue on GitHub for this so we don't lose sight of it? This is something that I am not as familiar with and want to make sure we get it documented correctly

Reply:
Follow-up item created #2780 marking this as resolved.

Reply:
This one was resolved by #2781